### ISSUE TITLE
what role -if any- do metrics play in the development process?

### ISSUE THREAD

**Issue by ahmadnassri** [OpenDevelopmentMethod/discussion#3]
question [asked by @bitergia](https://twitter.com/Bitergia/status/629165157848096768)

> What about adding development metrics to the methodology to keep process improving or at least controlled?

**Comment by jsmanrique** [comment]
Metrics should not "drive" a development process, but they can help on providing insights on what's going on in different areas:

* People / Community
  * Who is involved in the project?
  * Where people is working on (coding, reviewing, documenting)?
  * Where do they come from (internal contributors, companies, volunteers)?
  * What is their role (core, regular, occasional contributors)?
  * ...
* Activity
  * How much is being done (total activity, trends, etc.)?
  * Where is actvity focused on (analysis by project/repo/etc.)?
  * ...
* Effiency / Welness
  * Time to respond (issues, pull requests, etc.)
  * BMI (Backlog Management Indicator)
  * Retention rate (new comers or engaged vs still active over time)
  * ...

Those are just some ideas. Probably you could find more on [Evaluating Free / Open Source Software Projects](https://www.gitbook.com/book/jgbarah/evaluating-foss-projects/details)

**Comment by mchogan** [comment]
Does observation of open source projects indicate that metrics don't drive the process, or is that an opinion?

**Comment by louis-langholtz** [comment]
I don't believe a better development method can be developed without metrics. And we certainly can't scientifically defend a development methodology unless the metrics are things like: objective, repeatable, and reproducible. So I think metrics need to play a key role in the development process and as much as possible they should be scientifically meaningful. But coming up with such metrics for this seems less clear to me than the need to have them or the need to use them.

An important step seems to be defining what "better" needs to mean. Defining an elevator pitch sized primary objective of the "Open Development Method" also seems needed. Like: developing an optimal process for development. That sounds a lot like Six Sigma and/or Lean. I don't think these processes speak enough though to the human element: software developers, business leaders, customers, family, friends, etc. That's not to say that I don't appreciate Six Sigma or Lean. Only that I'd rather prefer it if we have a more narrow objective and a more human objective of something like: optimizing motivation in software development. I like this because then we can use my favorite theory of human motivation - Basic Psychological Needs Theory - as at least the initial scientific basis for the software development methodology. For more on this, please see [the issue I previously added](https://github.com/OpenDevelopmentMethod/discussion/issues/7).
